### PR TITLE
New Script: Google Maps search & directions

### PR DESCRIPTION
**Pull request by samstephenson**
## Description

New Script to search and get directions from Google Maps

## Type of change

- [x] New script command

## Screenshot

Screenshot 2021-02-18 at 14 14 03

## Dependencies / Requirements

None

## Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: commands/google-maps/google-maps.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# @raycast.argument1 { "type": "text", "placeholder": "Location"}
+# @raycast.argument2 { "type": "text", "placeholder": "Destination", "optional": true}
+
+if [$1 = false]; then
+	open "https://www.google.com/maps"
+elif [$2 = false]; then
+	open "https://www.google.com/maps/search/${1// /+}"
+else
+	open "https://www.google.com/maps/dir/${2// /+}/${1// /+}"
+fi
```
Comment (unnamedd) [inline commands/google-maps/google-maps.sh:20]: a small suggestion:

```suggestion
first_argument=${1// /+}
second_argument=${2// /+}

if [$1 = false]; then
 	open "https://www.google.com/maps"
 elif [$2 = false]; then
 	open "https://www.google.com/maps/search/$first_argument"
 else
 	open "https://www.google.com/maps/dir/$second_argument/$first_argument"
 fi
```

**Review by unnamedd — COMMENTED** [review]
So much simpler now @samstephenson, great conversion!  

The code looks great, I'm just making a suggestion here, but you can ignore if you prefer.

Thanks a lot for your contribution.

**Comment by samstephenson** [comment]
Thanks for your changes! Much better
Re-tested this a bit and realised the conditional wasn't working as it should so just fixed that

File: commands/google-maps/google-maps.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Optional parameters:
+# @raycast.icon google-map.png
+# @raycast.argument1 { "type": "text", "placeholder": "Location", "optional": true}
+# @raycast.argument2 { "type": "text", "placeholder": "Destination", "optional": true}
+
+first_argument=${1// /+}
+second_argument=${2// /+}
+
+if [$1 == ""]; then
```
Comment (PSalant726) [inline commands/google-maps/google-maps.sh:17]:
```suggestion
if [[ -z $1 ]]; then
```

File: commands/google-maps/google-maps.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	open "https://www.google.com/maps"
+elif [$2 == ""]; then
```
Comment (PSalant726) [inline commands/google-maps/google-maps.sh:19]:
```suggestion
elif [[ -z $2 ]]; then
```

File: commands/google-maps/google-maps.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# @raycast.mode silent
+# @raycast.packageName Google Maps
+
+# Optional parameters:
+# @raycast.icon google-map.png
+# @raycast.argument1 { "type": "text", "placeholder": "Location", "optional": true}
+# @raycast.argument2 { "type": "text", "placeholder": "Destination", "optional": true}
+
+first_argument=${1// /+}
+second_argument=${2// /+}
```
Comment (PSalant726) [inline commands/google-maps/google-maps.sh:15]: I think the naming of these variables could be a bit more clear/intuitive:
```suggestion
origin=${1// /+}
destination=${2// /+}
```

**Review by PSalant726 — APPROVED** [review]
Thanks for this, seems really useful! I just had some nitpicky suggestions, but nothing I'd say blocks a merge here. FYI - modifying your conditionals per my suggestions would be slightly more efficient and better align with bash norms.

File: commands/google-maps/google-maps.sh (hunk 4 above)
Comment (unnamedd) [reply commands/google-maps/google-maps.sh:15]: Much, much better than my suggestion! 😅

**Comment by PitNikola** [comment]
Looks good, gonna merge it 👍. @samstephenson, feel free to submit another PR if you want to apply suggestions from Phil.

**Pull request merged**